src/main/java/com/redhat/devtools/lsp4ij/features/formatting/LSPImprovedTextMateNestedBracesTypedHandler.java

```diff
@@ -11,26 +11,34 @@
 
 package com.redhat.devtools.lsp4ij.features.formatting;
 
+import com.intellij.codeInsight.CodeInsightSettings;
 import com.intellij.codeInsight.editorActions.TypedHandlerDelegate;
 import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.editor.ex.EditorEx;
 import com.intellij.openapi.project.Project;
 import com.intellij.psi.PsiFile;
+import com.redhat.devtools.lsp4ij.LSPIJEditorUtils;
+import com.redhat.devtools.lsp4ij.server.definition.launching.UserDefinedEditorBehaviorFeature;
 import org.jetbrains.annotations.NotNull;
+import org.jetbrains.plugins.textmate.editor.TextMateEditorUtils;
+import org.jetbrains.plugins.textmate.editor.TextMateTypedHandler;
+import org.jetbrains.plugins.textmate.language.preferences.TextMateAutoClosingPair;
+import org.jetbrains.plugins.textmate.language.syntax.lexer.TextMateScope;
 
 /**
  * Improves on-type handling of unbalanced nested brace/bracket/paren pairs in TextMate files.
  */
-public class LSPImprovedTextMateTypedHandler extends TypedHandlerDelegate {
+public class LSPImprovedTextMateNestedBracesTypedHandler extends TypedHandlerDelegate {
 
     @Override
     @NotNull
     public Result charTyped(char charTyped,
                             @NotNull Project project,
                             @NotNull Editor editor,
                             @NotNull PsiFile file) {
-        /* TODO: Uncomment once LSP4IJ no longer supports 2023.2
         if (LSPIJEditorUtils.isSupportedTextMateFile(file) &&
-            CodeInsightSettings.getInstance().AUTOINSERT_PAIR_BRACKET) {
+            CodeInsightSettings.getInstance().AUTOINSERT_PAIR_BRACKET &&
+            UserDefinedEditorBehaviorFeature.enableTextMateNestedBracesImprovements(file)) {
             int offset = editor.getCaretModel().getOffset();
             if (offset > 1) {
                 TextMateScope scopeSelector = TextMateEditorUtils.getCurrentScopeSelector((EditorEx) editor);
```

Comment:
Again, this check is safe because we've already determined that this file is a TextMate file, so we know the plugin is enabled before using its classes.